### PR TITLE
Add config to PrimTorch's nvFuser executor

### PR DESCRIPTION
**Pull request by IvanYashchuk**
This PR adds `executor_parameters` keyword argument to `torch._prims.executor.execute`.

For now there are two knobs:
* `use_python_fusion_cache: bool = True` whether to use lru_cache when constructing fusion object or not.
* `allow_single_op_fusion: bool = True` whether to allow fusions with single callable

Behavior can be controlled by passing dict with custom specified values as `executor_parameters` argument.

### CODE REVIEW

File: torch/_prims/nvfuser_executor.py [hunk 1]
```diff
@@ -209,18 +229,35 @@ def maybe_partition_graph(gm: GraphModule):
...
         return gm, any_unsupported
 
 
-def nvfuser_execute_partitioned(gm: GraphModule, *args):
+def nvfuser_execute_partitioned(gm: GraphModule, *args, executor_parameters=None):
+    executor_parameters = executor_parameters or DEFAULT_NVFUSER_PYTHON_CONFIG
+    # maybe_partition_graph function is cached so we can't use non-hashable arguments
+    allow_single_op_fusion = executor_parameters.get(
+        "allow_single_op_fusion",
+        DEFAULT_NVFUSER_PYTHON_CONFIG["allow_single_op_fusion"],
```
Comment (jjsjann123) [inline torch/_prims/nvfuser_executor.py:247]: why do we need a default value here? wasn't `executor_parameters` already a union with the default?

**Review by jjsjann123 — APPROVED** [review]
LGTM

File: torch/_prims/nvfuser_executor.py (hunk 1 above)
Comment (IvanYashchuk) [reply torch/_prims/nvfuser_executor.py:247]: If someone passes an empty dictionary or a dictionary with parameters omitting `allow_single_op_fusion` we'd still use the default and not error out. Calling `torch._prims.executor.execute(graph_module, args, executor_parameters={"none": None})` is allowed.
I should add tests for that.

`executor_parameters` is not a union with the default here, should it be? If `executor_parameters` is not specified or None then `executor_parameters==DEFAULT_NVFUSER_PYTHON_CONFIG`, if it's not None then `executor_parameters` is whatever was passed in by the user.

Comment (jjsjann123) [reply torch/_prims/nvfuser_executor.py:287]: "executor_parameters = executor_parameters or DEFAULT_NVFUSER_PYTHON_CONFIG"

Sorry I thought `or` here is a union `|`... :facepalm:

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**